tests/unittest/_torch/weight_sharing/test_post_transform_profiles.py

```
@@ -209,6 +243,201 @@ def test_explicitly_supported_optional_feature_is_qualified() -> None:
+        transfer_scope=PostTransformTransferScope.TARGET_MODEL,
+        runtime_config=_runtime_config(dtype="float16"),
+    )
+
+    assert decision.qualified
+    assert decision.profile is fp16_profile
+
+
+def test_registry_rejects_overlapping_runtime_profiles() -> None:
+    with pytest.raises(
+        ValueError,
+        match="Duplicate post-transform profile for .*runtime constraints overlap",
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)